### PR TITLE
feat: Upgrade kustomize to v5

### PR DESCRIPTION
**Pull request by Tomasz-Smelcerz-SAP**
**Description**

Upgrade to kustomize v5 and prepare for migration to Kustomization API version `v1`

Note: the `vars` configuration is also deprecated, but it is not migrated in this PR - on purpose. I wanted to avoid too many changes to the already complicated kustomize configurations we have. Once this PR is merged, `vars` will be migrated in the following PR.

Changes proposed in this pull request:

- upgrade `kustomize` to `v5.1.1` and `controller-gen` to `0.13.0`
- migrate deprecated customizations: `patchesJson6902` and `patchesStrategicMerge` are moved to the `patches` configuration.
- ensure the files render in the same way as in the current `main` branch.

**Related issue(s)**
https://github.com/kyma-project/lifecycle-manager/issues/391

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -116,7 +116,7 @@ uninstall: manifests kustomize ## Uninstall CRDs from the K8s cluster specified
 	$(KUSTOMIZE) build config/crd | kubectl delete --ignore-not-found=$(ignore-not-found) -f -
 
 .PHONY: deploy
-deploy: install manifests kustomize ## Deploy controller to the K8s cluster specified in ~/.kube/config.
```
Comment (lindnerby) [inline Makefile:119]: Can you check but I think `install` can also be removed from other targets, like `local-deploy-with-watcher`?

File: config/manager/kustomization.yaml [hunk 2]
```diff
@@ -15,5 +15,4 @@ configMapGenerator:
 
 images:
 - name: controller
-  newName: europe-docker.pkg.dev/kyma-project/prod/lifecycle-manager
```
Comment (lindnerby) [inline config/manager/kustomization.yaml:18]: I don't understand this change, can you please explain how this still works?

**Review by lindnerby — COMMENTED** [review]
Just left two questions. In general the changes look good of course.

File: config/manager/kustomization.yaml (hunk 2 above)
Comment (Tomasz-Smelcerz-SAP) [reply config/manager/kustomization.yaml]: Good catch! This sneaked in from this Makefile command: `cd config/manager && $(KUSTOMIZE) edit set image controller=${IMG}`

**Comment by Tomasz-Smelcerz-SAP** [comment]
Holding the PR until the changes are verified against control-plane kustomizations.

**Review by lindnerby — APPROVED** [review]

**Pull request merged**